### PR TITLE
feat: add `d/network_pool`

### LINKED ISSUE #199: Add support for `d/vcf_network_pool`
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Add support for `d/vcf_network_pool`.

If a network pool named "_eng_pool_" is already present you cannot use the following to retrieve the details of the network pool.

```hcl    
  data "vcf_network_pool" "eng_pool" {
       name = "eng_pool"
  }   
```

If we use this block, it fails as below:
  
```console  
Error: Invalid data source

on main.tf line 7, in data "vcf_network_pool" "eng_pool":
7: data "vcf_network_pool" "eng_pool" {

The provider vmware/vcf does not support data source "vcf_network_pool".

Did you intend to use the managed resource type "vcf_network_pool"? If so, declare this using a "resource" block instead of a "data" block.
```

### Use Case(s)

It will be used when data needs to be retrieved from an existing VCF system .

### Potential Configuration

```hcl
  data "vcf_network_pool" "eng_pool" {
       name = "eng_pool"
  }   
```

### References

_No response_

### PR DESCRIPTION
**Pull request by tenthirtyam**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/terraform-provider-vcf/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- Adds network pool data source: `data/vcf_network_pool`.
- Add documentation for the network pool datasource.
- Adds example for the network pool datasource.
- Removed duplicate `d/vcf_certificates` example.

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #199

**Test and Documentation Coverage**

For bug fixes or features:

- [x] Tests have been completed.
- [x] Documentation has been added/updated.

```console
Running tool: /opt/homebrew/bin/go test -timeout 30s -run ^TestAccDataSourceVcfNetworkPool$ github.com/vmware/terraform-provider-vcf/internal/provider

ok  	github.com/vmware/terraform-provider-vcf/internal/provider	0.440s
```

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

**Review by spacegospod — DISMISSED** [review]
LGTM

File: internal/provider/provider.go [hunk 1]
```diff
@@ -74,27 +74,28 @@ func Provider() *schema.Provider {
 		},
 
 		DataSourcesMap: map[string]*schema.Resource{
-			"vcf_domain":      DataSourceDomain(),
-			"vcf_cluster":     DataSourceCluster(),
-			"vcf_credentials": DataSourceCredentials(),
+			"vcf_cluster":      DataSourceCluster(),
+			"vcf_domain":       DataSourceDomain(),
+			"vcf_credentials":  DataSourceCredentials(),
```
Comment (dimitarproynov) [inline internal/provider/provider.go:79]: Why these changes? It is best to leave refactoring changes to a separate MR, not to pollute the current one with changes/complexity

File: examples/data-sources/cluster/vcf_credentials.tf [hunk 2]
```diff
@@ -1,15 +0,0 @@
...
-  }
-}
-
-data "vcf_credentials" "creds" {
-
-}
-
-data "vcf_credentials" "vcenter_credentials" {
-  resource_type = "VCENTER"
-}
```
Comment (dimitarproynov) [inline examples/data-sources/cluster/vcf_credentials.tf:15]: Why is this removed?

File: internal/provider/data_network_pool.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	d.SetId(networkPool.ID)
+	_ = d.Set("name", networkPool.Name)
+	_ = d.Set("networks", flattenNetworks(networkPool.Networks))
+
+	return nil
+}
+
+func GetNetworkPoolByName(ctx context.Context, apiClient *client.VcfClient, name string) (*models.NetworkPool, error) {
```
Comment (dimitarproynov) [inline internal/provider/data_network_pool.go:112]: Do you need this fuction to be public? Aka starts with capital letter

File: internal/provider/data_network_pool.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	params := network_pools.NewGetNetworkPoolParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+
+	networkPoolsPayload, err := apiClient.NetworkPools.GetNetworkPool(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if networkPoolsPayload.Payload == nil {
+		return nil, errors.New("error: network pool not found")
```
Comment (dimitarproynov) [inline internal/provider/data_network_pool.go:122]: I don't think you need the "error:" prefix

File: internal/provider/data_network_pool.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	for _, networkPool := range networkPoolsPayload.Payload.Elements {
+		if networkPool.Name == name {
+			return networkPool, nil
+		}
+	}
+
+	return nil, errors.New("error: network pool not found")
```
Comment (dimitarproynov) [inline internal/provider/data_network_pool.go:131]: same as above

**Review by dimitarproynov — CHANGES_REQUESTED** [review]

File: examples/data-sources/cluster/vcf_credentials.tf (hunk 2 above)
Comment (tenthirtyam) [reply examples/data-sources/cluster/vcf_credentials.tf:15]: this was a duplicate entry in the wrong folder. It's noted in the PR details.

File: internal/provider/data_network_pool.go (hunk 3 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go:112]: Great point!

File: internal/provider/data_network_pool.go (hunk 4 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go:122]: I'll adjust.

File: internal/provider/data_network_pool.go (hunk 5 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go:131]: I'll adjust this.

File: internal/provider/provider.go (hunk 1 above)
Comment (tenthirtyam) [reply internal/provider/provider.go]: This was cleared out in the rebase and push in 695b133.

File: internal/provider/data_network_pool.go (hunk 4 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go]: Addressed in 60f13a9.

File: internal/provider/data_network_pool.go (hunk 5 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go]: Addressed in 60f13a9.

File: internal/provider/data_network_pool.go (hunk 3 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go]: Addressed in 60f13a9.

File: internal/provider/data_network_pool.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+	params := network_pools.NewGetNetworkPoolParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+
+	networkPoolsPayload, err := apiClient.NetworkPools.GetNetworkPool(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if networkPoolsPayload.Payload == nil {
+		return nil, errors.New(" network pool not found")
```
Comment (dimitarproynov) [inline internal/provider/data_network_pool.go:122]: nit: you can remove the leading space

File: internal/provider/data_network_pool.go [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	for _, networkPool := range networkPoolsPayload.Payload.Elements {
+		if networkPool.Name == name {
+			return networkPool, nil
+		}
+	}
+
+	return nil, errors.New(" network pool not found")
```
Comment (dimitarproynov) [inline internal/provider/data_network_pool.go:131]: nit: same as above

**Review by dimitarproynov — DISMISSED** [review]
LGTM!

File: internal/provider/data_network_pool.go (hunk 7 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go]: Resoved in 6fd9432.

**Review by dimitarproynov — APPROVED** [review]

File: internal/provider/data_network_pool.go (hunk 6 above)
Comment (tenthirtyam) [reply internal/provider/data_network_pool.go]: Resoved in 6fd9432.
